src/button/Link.tsx

```diff
@@ -38,15 +44,32 @@ export function useLink(
 type Props = Omit<
   ComponentPropsWithoutRef<typeof CpdLink>,
   "href" | "onClick"
-> & { to: LocationDescriptor };
+> & { to: LocationDescriptor; state?: unknown };
 
 /**
  * A version of Compound's link component that integrates with our router setup.
  */
 export const Link = forwardRef<HTMLAnchorElement, Props>(function Link(
-  { to, ...props },
+  { to, state, ...props },
   ref,
 ) {
-  const [path, onClick] = useLink(to);
+  const [path, onClick] = useLink(to, state);
   return <CpdLink ref={ref} {...props} href={path} onClick={onClick} />;
 });
+
+export const ExternalLink = forwardRef<
```

Comment:
I think a guiding comment/docstring here clarifying in what cases this should be used also would be nice.

Or maybe even a docstring for all link components listing what other optiosn there are so the devs stumble over a docsting that mentions that there are other types they might want to/need to consider.